indexer/src/main.rs

```diff
@@ -21,6 +22,9 @@ async fn handle_message(
     strict_mode: bool,
     receipts_cache: receipts_cache::ReceiptsCache,
 ) -> anyhow::Result<()> {
+    metrics::BLOCK_COUNT.inc();
+    metrics::BLOCK_HEIGHT.set(streamer_message.block.header.height.try_into().unwrap());
```

Comment:
Just FYI here we cast from u64 to i64
We can say that block height will never be bigger than i64::MAX, but it's better not to play such games and follow the types strictly

Reply:
Thanks for the heads up!

I've added a new metrics which exposes `block.header.timestamp_nanosec` which is also `u64`, is it safe to cast this as `i64`? If not, is there another prometheus metrics type I can use?

Reply:
It's always not safe to cast u64 to i64. 
I haven't researched prometheus supported types. 
Please re-check this yourself, I trust you. If i64 is the only/the best type that we can use - let's go with it and add the comment about it.
I'd also prefer to avoid using `unwrap`, it will give us unreadable error. We can go with `.map_err()?` and provide some context there in addition to the comment.
cc @roshaans

Reply:
If this metric type was a counter, then we would not have a problem as counter supports u64 types, but we need this metric to be a gauge type because block height does not always go up all the time (in case of restarting the indexer)

there are two types of gauges. `IntGauge` uses an i64 type and a simple `gauge` uses f64 type. 

we can opt into using the f64 type, but that comes with performance trade-offs. See [https://github.com/tikv/rust-prometheus/issues/149#issuecomment-369497385](here.)

I say we stick with IntGauge till the community comes up with a gauge that supports u64 types.

Reply:
It's not only about number of days, it's also about the data we have. In this field, we can receive any value with u64 type and we should be able to handle it properly. So, if the library does not support u64, which is sad, we should at least provide the doc and not try to hide the issue

Reply:
> We are probably safe for another 1.0675199e+14 days if blocks are produced every second of the day

@roshaans how did you come to that conclusion? I end up with something much lower:
```
// i164::MAX - latest block timestamp = remaining nanoseconds
9223372036854775807 - 1669860597023961766 = 7,553,511,439,830,814,041

// remaining nanoseconds / number of nanoseconds in second = remaining seconds
7553511439830814041 / 1000000000 = 7,553,511,439

// remaining seconds -> days = remaining days
7553511439 / 60 / 60 / 24 = 87,424
```
If my calculations are correct there are `87,724` days or `240` years. This still seems plenty but I agree the concerns @telezhnaya is raising.

The goal of exposing this timestamp was to get a time diff/latency from the tip of the network. But given that nearcore only exposes block metrics in `height` rather than `nanos`, probably due to the same concerns we have here, we don't have anything to compare to.

I've been playing around with Grafana and realised we can get latency without `LATEST_BLOCK_TIMESTAMP`. Using `LATEST_BLOCK_HEIGHT` to get the height difference from the tip of the network, and dividing that by the rate of change of `BLOCK_HEIGHT`, we get the delay of the indexer from the tip. I've already set this up for `near-lake-indexer`, you can see it [here](https://nearinc.grafana.net/d/z-ssj7F4z/lake-indexer?orgId=1&var-chain=testnet&var-chain_id=testnet&refresh=5s).

**tldr;**
Given the concerns we have about `i64`/`u64`, and the workaround possible without `LATEST_BLOCK_TIMESTAMP`, I'm going to remove it.